src/lib/draw/StreetViewMode.svelte

```diff
@@ -0,0 +1,131 @@
+<script lang="ts">
+  import { onDestroy } from "svelte";
+  import type { Mode } from "./types";
+  import type { GeoJSONSource, MapMouseEvent } from "maplibre-gl";
+  import type { Feature, Point } from "geojson";
+  import { point } from "@turf/helpers";
+  import { map } from "../../stores";
+  import {
+    emptyGeojson,
+    overwriteSource,
+    overwriteLayer,
+    drawCircle,
+  } from "../../maplibre_helpers";
+
+  const thisMode = "street-view";
+
+  const circleRadiusPixels = 10;
+
+  export let mode: Mode;
+  export let changeMode: (m: Mode) => void;
+
+  let imagery: "google" | "bing" | "mapillary" = "google";
+
+  export function start() {}
+  export function stop() {
+    cursor = null;
+  }
+
+  let cursor: Feature<Point> | null = null;
+
+  $map.on("mousemove", onMouseMove);
+  $map.on("click", onClick);
+
+  onDestroy(() => {
+    $map.off("mousemove", onMouseMove);
+    $map.off("click", onClick);
+  });
+
+  // Rendering
+  let source = "street-view";
+  overwriteSource($map, source, emptyGeojson());
+  // TODO Different icon?
+  overwriteLayer($map, {
+    id: "draw-street-view",
+    source,
+    ...drawCircle("black", circleRadiusPixels, 1.0),
+  });
+
+  $: {
+    let gj = emptyGeojson();
+    if (cursor) {
+      gj.features.push(cursor);
+    }
+    ($map.getSource(source) as GeoJSONSource).setData(gj);
+  }
+
+  function onMouseMove(e: MapMouseEvent) {
+    if (mode != thisMode) {
+      return;
+    }
+
+    cursor = cursorFeature(e.lngLat.toArray());
+  }
+
+  function onClick() {
+    if (mode != thisMode) {
+      return;
+    }
+
+    let [lon, lat] = cursor.geometry.coordinates;
+    if (imagery == "google") {
+      window.open(
```

Comment:
The URL is mostly self explanatory, but some magic params lifted from https://nptscot.github.io/#8/55.953/-3.138. I couldn't find a clear guide anywhere explaining these params and letting us, for example, always start facing north